docs/guide.md

```diff
@@ -0,0 +1,286 @@
+	// deducted before the Msg is processed.
+	GetFeePayer() crypto.Address
+
+	// Get the canonical byte representation of the Tx.
+	// Includes any signatures (or empty slots).
+	GetTxBytes() []byte
+
+	// Signatures returns the signature of signers who signed the Msg.
+	// CONTRACT: Length returned is same as length of
+	// pubkeys returned from MsgKeySigners, and the order
+	// matches.
+	// CONTRACT: If the signature is missing (ie the Msg is
+	// invalid), then the corresponding signature is
+	// .Empty().
+	GetSignatures() []StdSignature
+}
+```
+
+The `tx.GetSignatures()` method returns a list of signatures,
+which must match the list of addresses returned by
+`tx.Msg.GetSigners()`. The signatures come in a standard form:
+
+```golang
+type StdSignature struct {
+	crypto.PubKey // optional
+	crypto.Signature
+	Sequence int64
+}
+```
+
+It contains the signature itself, as well as the corresponding
+account's sequence number.  The sequence number is expected to
+increment every time a message is signed by a given account.
+This prevents "replay attacks", where the same message could be
+executed over and over again.
+
+The `StdSignature` can also optionally include the public key for
+verifying the signature.  An application can store the public key
+for each address it knows about, making it optional to include
+the public key in the transaction. In the case of Basecoin, the
+public key only needs to be included in the first transaction
+send by a given account - after that, the public key is forever
+stored by the application and can be left out of transactions.
+
+Transactions can also specify the address responsible for paying
+the transaction's fees using the `tx.GetFeePayer()` method.
+
+The standard way to create a transaction from a message is to use
+the `StdTx`: 
+
+```golang
+type StdTx struct {
+	Msg
+	Signatures []StdSignature
+}
+```
+
+### Encoding and Decoding Transactions
+
+Messages and transactions are designed to be generic enough for
+developers to specify their own encoding schemes.  This enables
+the SDK to be used as the framwork for constructing already
+specified cryptocurrency state machines, for instance Ethereum. 
+
+When initializing an application, a developer must specify a
+`TxDecoder` function which determines how an arbitrary byte array
+should be unmarshalled into a `Tx`: 
+
+```golang
+type TxDecoder func(txBytes []byte) (Tx, error)
+```
+
+In `Basecoin`, we use the Tendermint wire format and the
+`go-wire` library for encoding and decoding all message types.
+The `go-wire` library has the nice property that it can unmarshal
+into interface types, but it requires the relevant types to be
+registered ahead of type. Registration happens on a `Codec`
+object, so as not to taint the global name space.
+
+For instance, in `Basecoin`, we wish to register the `SendMsg`
+and `IssueMsg` types:
+
+```golang
+cdc.RegisterInterface((*sdk.Msg)(nil), nil)
```

Comment:
What happens if two modules use the same name and hence have the same prefix bytes?

Reply:
They shouldn't... worst case scenario you need to version these names, like "Msg.v1" and "Msg.v2".  Regardless. you might end up with two different names that have the same prefix bytes.  In that case, wire will tell you what to do.  Namely, wire will write the disambiguation bytes.